packages/storage/src/utils/CursorBuffer.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * A wrapper around byte buffers to perform cursor reading on bytes
+ * of different sizes
+ *
+ */
+export class CursorBuffer {
+  private readonly dataView: DataView;
+
+  private readonly littleEndian: boolean;
+
+  private bytePositon: number;
+
+  constructor(typedArray: Uint8Array, littleEndian = true) {
+    this.dataView = new DataView(
+      typedArray.buffer,
+      typedArray.byteOffset,
+      typedArray.byteLength,
+    );
+    this.littleEndian = littleEndian;
+    this.bytePositon = 0;
+  }
+
+  /**
+   * Reads 1 byte
+   *
+   */
+  public read8(): number {
+    const value = this.dataView.getUint8(this.bytePositon);
+    this.bytePositon += 1;
```

Comment:
Does this not need the endian value?

Reply:
No it doesn't. Since endianness affects layout of each consecutive byte blocks so a simple 8bit read (1 byte) would always be in the correct order.

Reply:
Got it, that makes, thank you. Was wondering if it was by bit 😅